crates/client/src/lib.rs

```diff
@@ -310,109 +328,173 @@ impl<R: RegistryStorage, C: ContentStorage, N: NamespaceMapStorage> Client<R, C,
+
+        #[cfg(feature = "cli-interactive")]
+        let mut accepted_prompt_to_initialize = false;
+
+        let mut init_record_id: Option<RecordId> = None;
+
+        let (package, record) = loop {
+            let mut info = publish_info.clone();
+
+            #[cfg(feature = "cli-interactive")]
+            let mut initializing = info.initializing();
+
+            #[cfg(not(feature = "cli-interactive"))]
+            let initializing = info.initializing();
+
+            let package = match self.fetch_package(&info.name).await {
+                Ok(package) => {
+                    if initializing {
+                        return Err(ClientError::CannotInitializePackage {
+                            name: package.name,
+                            init_record_id,
+                        });
+                    } else if info.head.is_none() {
+                        // If we're not initializing the package and a head was not explicitly specified,
+                        // set to the latest known head.
+                        info.head = package.state.head().as_ref().map(|h| h.digest.clone());
+                    }
+                    package
                 }
-                package
-            }
-            Err(ClientError::PackageDoesNotExist {
-                name,
-                has_auth_token,
-            }) => {
-                if !initializing {
-                    let prompt = if has_auth_token {
-                        format!(
-"Package `{package_name}` does not already exist or you do not have access.
-Do you wish to initialize `{package_name}` and publish release y/N\n",
-package_name = &info.name,
-)
-                    } else {
-                        format!(
-"Package `{package_name}` does not already exist or you do not have access.
-You may be required to login. Try: `warg login`
-Do you wish to initialize `{package_name}` and publish release y/N\n",
-package_name = &info.name,
-)
-                    };
-                    if Confirm::with_theme(&ColorfulTheme::default())
-                        .with_prompt(prompt)
-                        .default(false)
-                        .interact()
-                        .unwrap()
-                    {
-                        info.entries.insert(0, PublishEntry::Init);
-                    } else {
+                Err(ClientError::PackageDoesNotExist {
+                    name,
+                    has_auth_token,
+                }) => {
+                    if !initializing && cfg!(not(feature = "cli-interactive")) {
                         return Err(ClientError::MustInitializePackage {
                             name,
                             has_auth_token,
                         });
+                    } else if !initializing {
+                        #[cfg(feature = "cli-interactive")]
+                        {
+                            if self.disable_interactive {
```

Comment:
You could probably move relevant `use`s down here, e.g.

```suggestion
                        #[cfg(feature = "cli-interactive")]
                        {
                            use dialoguer::{theme::ColorfulTheme, Confirm};
                            if self.disable_interactive {
```

Reply:
Simplified